cvat-data/src/js/cvat-data.js

```diff
@@ -116,55 +119,95 @@ class FrameProvider {
+           
+        }else{
+            let zip = new JSZip();
+            let that = this;
+            zip.loadAsync(block).then(function(_zip){
+                let index = start;
+               
+                _zip.forEach(function (relativePath, zipEntry) { 
+                    
+                   _zip.file(relativePath).async('ArrayBuffer').then(function (fileData){                             
+                          that._frames[index] = jpeg.decode(fileData); 
+                          index ++;
+                          callback(index - 1);
+                    });
+                });
+            });
+
+        }
+    }
+
+    /*
+        Method decodes a block of data
+
+        @param start {number} - is the first frame of a block
+        @param end {number} - is the last frame of a block + 1
+        @param callback - callback)
 
-            try {
-                this._running = true;
-                this._blocks.push(`${start}:${end}`);
-
-                this._cleanup();
-
-                if (this._blockType === BlockType.TSVIDEO) {
-                    this._demuxer.write(block);
-                    for (let i = start; i < end; i++) {
-                        const result = this._videoDecoder.decode();
-                        if (!Array.isArray(result)) {
-                            const message = 'Result must be an array.'
-                                + `Got ${result}. Possible reasons: `
-                                + 'bad video file, unpached jsmpeg';
-                            throw Error(message);
-                        }
-                        this._frames[i] = YCbCrToRGBA(...result);
-                    }
-                    resolve();
-                } else if (this._blockType === BlockType.ARCHIVE) {
-                    // to do with any npm dearchiver
-                }
-            } catch (error) {
-                reject(error);
-            } finally {
-                this._running = false;
+    */
+    decode(end, callback) {
+
+        try{
+            
+            const result = this._videoDecoder.decode();
+            if (!Array.isArray(result)) {
+                clearTimeout(this._decode_id);
+                const message = 'Result must be an array.'
+                    + `Got ${result}. Possible reasons: `
+                    + 'bad video file, unpached jsmpeg';
+                throw Error(message);
             }
+            this._frames[this._currFrame] = YCbCrToRGBA(...result);
+            this._currFrame ++;            
+            callback(this._currFrame - 1);
+           
+            if (this._currFrame != end)
+            {
+                this._decode_id = setTimeout(this.decode.bind(this, end, callback), 10);
+            } else {
+                clearTimeout(this._decode_id);
+            }
+        } catch(error) {
+            clearTimeout(this._decode_id);
```

Comment:
You don't need clearTimeout() after they triggered. 
So, in this case with timeouts you can remove all clearTimeout calls.